src/Medidata.ZipkinTracer.Core/SpanTracer.cs

```diff
@@ -81,11 +81,11 @@ public virtual Span SendClientSpan(string requestName, string traceId, string pa
             return newSpan;
         }
 
-        public virtual void ReceiveClientSpan(Span span, string clientServiceName)
+        public virtual void ReceiveClientSpan(Span span)
         {
             var annotation = new Annotation()
             {
-                Host = zipkinEndpoint.GetEndpoint(clientServiceName),
+                Host = span.Annotations[0].Host,
```

Comment:
@bvillanueva-mdsol Thanks, now I see the null checks in the `ZipkinClient` (sorry I missed it at the first time).

Regarding the non-empty `Annotations`, I understand that if the `ZipkinClient.EndClientTrace()` will be called with non-conform `clientSpan` (that is, with null/empty `Annotations`) the `SpanTracer.ReceiveClientSpan()` will throw an exception which will be logged (i.e. handled) and this won't cause any problems in our case, but one advantage of handling this case in the `ReceiveClientSpan()` method is that you could have a great chance to throw a more specific exception (with a descriptive message) for the log: instead of getting an `IndexOutOfRangeException` with some generic message, the reader of the logs could see either an `ArgumentException` or `InvalidOperationException` indicating with a message that the `Annotations` is empty and this way it is not possible to get the host.

But of course I would not consider this a must-have option just a recommendation so I am leaving the decision to you ;).

Btw I could not see any enforcement by the client to call the `EndClientTrace()` method with the same `clientSpan` which produced by the `StartClientTrace()` method, so passing such span mentioned above would be possible.